src/actions/hubspot/hubspot.ts

```diff
@@ -0,0 +1,257 @@
+    tag,
+  }: DefaultHubspotConstructorProps) {
+    super()
+    this.name = name
+    this.label = label
+    this.description = description
+    this.call = call
+    this.tag = tag
+    this.requiredFields = [{ any_tag: [tag] }]
+  }
+  supportedFormats = (request: Hub.ActionRequest) => {
+    if (request.lookerVersion && semver.gte(request.lookerVersion, "6.2.0")) {
+      return [Hub.ActionFormat.JsonDetailLiteStream]
+    } else {
+      return [Hub.ActionFormat.JsonDetail]
+    }
+  }
+
+  async execute(request: Hub.ActionRequest) {
+    return this.executeHubspot(request)
+  }
+
+  protected hubspotClientFromRequest(request: Hub.ActionRequest) {
+    return new hubspot.Client({ apiKey: request.params.hubspot_api_key })
+  }
+
+  protected taggedFields(fields: Hub.Field[], tags: string[]) {
+    return fields.filter(
+      (f) =>
+        f.tags &&
+        f.tags.length > 0 &&
+        f.tags.some((t: string) => tags.indexOf(t) !== -1),
+    )
+  }
+
+  protected getHubspotIdFieldName(fieldset: Hub.Field[]): string | undefined {
+    let fieldName: string | undefined
+    fieldset.forEach((field) => {
+      if (
+        field.tags &&
+        field.tags.length > 0 &&
+        field.tags.includes(this.tag)
+      ) {
+        fieldName = field.name
+      }
+    })
+    return fieldName
+  }
+
+  /**
+   * Returns the hubspot ID from the current row, given that one of the column dimensions
+   * was tagged with the corresponding HubspotTag
+   * @param fieldset Fieldset for the entire query
+   * @param row The specific row to be processed
+   */
+  protected getHubspotIdFromRow(
+    fieldset: Hub.Field[],
+    row: Hub.JsonDetail.Row,
+  ): string | undefined {
+    const hubspotIdFieldName = this.getHubspotIdFieldName(fieldset)
+    if (!hubspotIdFieldName) {
+      return undefined
+    }
+    for (const field of fieldset) {
+      if (field.name === hubspotIdFieldName) {
+        return row[hubspotIdFieldName].value
+      }
+    }
+  }
+
+  protected async executeHubspot(request: Hub.ActionRequest) {
+    const hubspotClient = this.hubspotClientFromRequest(request)
+
+    let hiddenFields: string[] = []
+    if (
+      request.scheduledPlan &&
+      request.scheduledPlan.query &&
+      request.scheduledPlan.query.vis_config &&
+      request.scheduledPlan.query.vis_config.hidden_fields
+    ) {
+      hiddenFields = request.scheduledPlan.query.vis_config.hidden_fields
+    }
+
+    let hubspotIdFieldName: string | undefined
+    const batchUpdateObjects: hubspot.contactsModels.SimplePublicObjectBatchInput[] = []
+    let fieldset: Hub.Field[] = []
+    const errors: Error[] = []
+
+    try {
+      await request.streamJsonDetail({
+        onFields: (fields) => {
+          fieldset = Hub.allFields(fields)
+          hubspotIdFieldName = this.getHubspotIdFieldName(fieldset)
+          if (!hubspotIdFieldName) {
+            const error = `Dimension with the ${this.tag} tag is required`
+            winston.error(error)
```

Comment:
if you can add the webhookID to the error and log messages it makes it a lot easier to debug issues when they pop up